src/script/api/script_text.hpp

```diff
@@ -21,17 +23,17 @@ class Text : public ScriptObject {
 public:
 	/**
 	 * Convert a ScriptText to a normal string.
-	 * @return A string (in a static buffer), or nullptr.
+	 * @return A string, or <empty>.
 	 * @api -all
 	 */
-	virtual const char *GetEncodedText() = 0;
+	virtual std::optional<const std::string> GetEncodedText() = 0;
```

Comment:
If we replace the special handling of SCRIPT_TEXT_MAX_PARAMETERS by some kind of assertion check, it's not even needed to use std::optional. The number of parameters is already checked in constructor (throws "invalid parameter"), and AddParam() or SetParam() can't add more than the limit (no throw, the call just fails and says "unknown error").

Maybe I should just throw in AddParam() and SetParam() to give useful info, and also find a way to kill the script in case it managed to add more than SCRIPT_TEXT_MAX_PARAMETERS (the special handling requiring std::optional), or just assume it can't happen, but I don't like removing a safety check.